### PR TITLE
Set part-type as optional when creating a partition

### PR DESCRIPTION
**Pull request by Baitinq**
The man page specifies that part-type may be specified only with msdos and  dvh  partition  tables, hence it shouldnt be used with gpt partitions. This change makes it able to do this by making this attribute optional

### CODE REVIEW

**Review by Mic92 — APPROVED** [review]

**Comment by Mic92** [comment]
We could also simplify our examples than...

**Comment by Lassulus** [comment]
Will merge this after my mega PR is done, maybe I will also just absorb it into there

**Comment by Baitinq** [comment]
@Lassulus Ill also implement partition labels support when ur pr is done:)

**Comment by Mic92** [comment]
> @Lassulus Ill also implement partition labels support when ur pr is done:)

This is now the case.

**Comment by Baitinq** [comment]
Ill rebase this against master as soon as #30 is merged.

**Comment by Mic92** [comment]
It does not seem to be an optional parameter: https://www.gnu.org/software/parted/manual/html_chapter/parted_2.html#SEC18. We could potentiality have a default value though.

File: default.nix
Comment (Mic92) [inline default.nix:110]:
```suggestion
    parted -s "''${${env}}" mkpart ${x.part-type or ""} ${x.fs-type or "primary"} ${x.start} ${x.end}
```

**Comment by Baitinq** [comment]
@Lassulus I've fixed the conflict with the master branch, should I squash the commits or can you do that while merging?

File: default.nix
Comment (Baitinq) [reply default.nix]: From the arch wiki (https://wiki.archlinux.org/title/Parted), part-type is only needed when using MBR.

Comment (Baitinq) [reply default.nix]: @Mic92 I believe the man page you have linked is outdated. This is the current parted man page which specifies the optionality of part-type (https://www.gnu.org/software/parted/manual/parted.html#mkpart)

**Comment by Lassulus** [comment]
this was absorbed into the types rewrite. thanks!

**Pull request closed without merging**
